$OpenBSD: patch-tools_clang_lib_Basic_Targets_X86_h,v 1.6 2020/11/18 21:20:00 jca Exp $

- implement -msave-args in clang/llvm, like the sun did for gcc.
- Disable strict floating point if not X86.

Index: tools/clang/lib/Basic/Targets/X86.h
--- tools/clang/lib/Basic/Targets/X86.h.orig
+++ tools/clang/lib/Basic/Targets/X86.h
@@ -123,6 +123,7 @@ class LLVM_LIBRARY_VISIBILITY X86TargetInfo : public T
   bool HasMOVDIR64B = false;
   bool HasPTWRITE = false;
   bool HasINVPCID = false;
+  bool HasSaveArgs = false;
   bool HasENQCMD = false;
 
 protected:
@@ -147,6 +148,7 @@ class LLVM_LIBRARY_VISIBILITY X86TargetInfo : public T
       : TargetInfo(Triple) {
     LongDoubleFormat = &llvm::APFloat::x87DoubleExtended();
     AddrSpaceMap = &X86AddrSpaceMap;
+    HasStrictFP = true;
   }
 
   const char *getLongDoubleMangling() const override {
